### PR TITLE
ui enhancements for fresheyes

### PR DESCRIPTION
**Pull request by IgboPharaoh**
- Add menu bar
- Add title for clickable cards section
- Show installed repositories for FreshEyes

### CODE REVIEW

**Comment by Extheoisah** [comment]
Two UI nits for which I'd like @aassoiants and @rejeses opinions (see screenshot attached for UI changes)
1. There is no indication that when you click on the avatar, it opens the popover. I was thinking of adding a small arrow icon to make it obvious that it's clickable?

2. What do you think of the copy above the suggestion cards?

image

File: client/fresheyes-web/src/app/assets/icons/exitArrow.tsx [hunk 1]
```diff
@@ -0,0 +1,21 @@
+import React from "react";
```
Comment (Extheoisah) [inline client/fresheyes-web/src/app/assets/icons/exitArrow.tsx:1]:
```suggestion
"use client"

import React from "react";
```
we should explicitly make this a client component

File: client/fresheyes-web/src/app/components/form/CustomComponents.tsx [hunk 2]
```diff
@@ -39,19 +39,22 @@ export const ClickableOptions = ({
...
       className={`h-full w-full dark:text-white min-h-[100px] min-w-[170px] p-3 rounded-xl md:hover:scale-105 flex flex-col items-start relative border-b border-gray-200 bg-white dark:bg-gradient-to-b from-zinc-200 backdrop-blur-2xl dark:border-neutral-800 dark:bg-zinc-800/30 dark:from-inherit lg:border lg:bg-gray-200 lg:dark:bg-zinc-800/30`}
       onClick={async () => {
         setFormValues((prev) => ({ ...prev, owner, repo, pull_number }));
-        await onClickOption({ owner, repo, pull_number });
       }}
     >
       <section className=' absolute right-3 bottom-3'>
         <p className='text-xs dark:text-white border-[1.5px] p-2 px-3 rounded-3xl font-bold'>RUN</p>
       </section>
+      <p className=' uppercase text-sm font-bold underline text-start pb-2'>{title}</p>
```
Comment (Extheoisah) [inline client/fresheyes-web/src/app/components/form/CustomComponents.tsx:57]:
```suggestion
      <p className='uppercase text-sm font-bold underline text-start pb-2'>{title}</p>
```
nit: there are several cases like this in the code. We can address it in another pr by setting up prettier and eslint

File: client/fresheyes-web/src/app/components/form/FormSection.tsx [hunk 3]
```diff
@@ -168,36 +173,31 @@ const FormSection = ({ username }: { username: string | null | undefined }) => {
       </div>
 
       <div className='flex flex-col gap-6 overflow-x-scroll max-w-5xl  xl:w-3/4 md:p-4 w-full'>
+        <h2 className=' text-base md:text-xl font-semibold'>Want to see how FreshEyes work ? Try cloning some of these repositories</h2>
```
Comment (Extheoisah) [inline client/fresheyes-web/src/app/components/form/FormSection.tsx:176]:
```suggestion
        <h2 className='text-base md:text-xl font-semibold'>Want to see how FreshEyes works? Try cloning some of these repositories</h2>
```

File: client/fresheyes-web/src/app/components/form/FormSection.tsx [hunk 4]
```diff
@@ -168,36 +173,31 @@ const FormSection = ({ username }: { username: string | null | undefined }) => {
...
         <section className='flex flex-col md:flex-row w-full gap-6'>
-          <ClickableOptions
-            owner={"bitcoin"}
-            repo={"bitcoin"}
-            pull_number={8149}
-            onClickOption={() => processPullRequest(true, { owner: "bitcoin", repo: "bitcoin", pull_number: 8149 })}
-            setFormValues={setFormValues}
-          />
+          <ClickableOptions owner={"bitcoin"} repo={"bitcoin"} pull_number={8149} setFormValues={setFormValues} title={"Segregated witness"} />
```
Comment (Extheoisah) [inline client/fresheyes-web/src/app/components/form/FormSection.tsx:178]:
```suggestion
          <ClickableOptions owner={"bitcoin"} repo={"bitcoin"} pull_number={8149} setFormValues={setFormValues} title={"Segregated witness (Segwit)"} />
```

File: client/fresheyes-web/src/app/components/form/FormSection.tsx [hunk 5]
```diff
@@ -168,36 +173,31 @@ const FormSection = ({ username }: { username: string | null | undefined }) => {
...
           />
         </section>
         <section className='flex flex-col md:flex-row w-full gap-6'>
           <ClickableOptions
             owner={"lightningnetwork"}
             repo={"lnd"}
             pull_number={2022}
-            onClickOption={() => processPullRequest(true, { owner: "lightningnetwork", repo: "lnd", pull_number: 2022 })}
             setFormValues={setFormValues}
+            title={"htlcswitch: hodl invoice"}
```
Comment (Extheoisah) [inline client/fresheyes-web/src/app/components/form/FormSection.tsx:193]:
```suggestion
            title={"Htlcswitch: hodl invoice"}
```
Let's be consistent with the alphabet casing

File: client/fresheyes-web/src/app/components/form/bot.ts [hunk 6]
```diff
@@ -15,6 +15,58 @@ const app = new App({
...
+      success: false,
+      list: null,
+    };
+  }
+
+  try {
+    const installations = await app.octokit.request("GET /app/installations");
+
+    for (const installation of installations.data) {
+      if (installation.account && installation.app_id === Number(process.env.GITHUB_APP_ID)) {
```
Comment (Extheoisah) [inline client/fresheyes-web/src/app/components/form/bot.ts:32]: `process.env.GITHUB_APP_ID` maybe null. We need to handle the error case of its null or undefined

File: client/fresheyes-web/src/app/components/form/bot.ts [hunk 7]
```diff
@@ -15,6 +15,58 @@ const app = new App({
...
+        const installationOctokit = await app.getInstallationOctokit(installation.id);
+
+        try {
+          const repos = await installationOctokit.request("GET /installation/repositories");
+          console.log(repos.data.repositories);
```
Comment (Extheoisah) [inline client/fresheyes-web/src/app/components/form/bot.ts:37]:
```suggestion
```

File: client/fresheyes-web/src/app/components/menu/Menu.tsx [hunk 8]
```diff
@@ -0,0 +1,66 @@
...
+import Image from "next/image";
+import { StaticImport } from "next/dist/shared/lib/get-img-props";
+import { signOut } from "next-auth/react";
+import { getInstalledRepositories } from "../form/bot";
+import { ExitArrowIcon } from "@/app/assets/icons/exitArrow";
+
+export const Menu = ({ image }: { image: string | StaticImport | null | undefined }) => {
+  const [open, setOpen] = useState(false);
+  const [loading, setLoading] = useState(false);
+  const [installedRepos, setinstalledRepos] = useState<any[] | null>([]);
```
Comment (Extheoisah) [inline client/fresheyes-web/src/app/components/menu/Menu.tsx:14]:
```suggestion
  const [installedRepos, setInstalledRepos] = useState<any[] | null>([]);
```
nit: use camelCase

File: client/fresheyes-web/src/app/components/menu/Menu.tsx [hunk 9]
```diff
@@ -0,0 +1,66 @@
...
+
+  React.useEffect(() => {
+    const getInstalledRepos = async () => {
+      setLoading(true);
+      setinstalledRepos([]);
```
Comment (Extheoisah) [inline client/fresheyes-web/src/app/components/menu/Menu.tsx:19]:
```suggestion
      setInstalledRepos([]);
```

File: client/fresheyes-web/src/app/components/menu/Menu.tsx [hunk 10]
```diff
@@ -0,0 +1,66 @@
...
+      const res = await getInstalledRepositories();
+      setinstalledRepos(res.list);
+      setLoading(false);
+      console.log(res);
```
Comment (Extheoisah) [inline client/fresheyes-web/src/app/components/menu/Menu.tsx:23]:
```suggestion
```

File: client/fresheyes-web/src/app/components/menu/Menu.tsx [hunk 11]
```diff
@@ -0,0 +1,66 @@
...
+  const [open, setOpen] = useState(false);
+  const [loading, setLoading] = useState(false);
+  const [installedRepos, setinstalledRepos] = useState<any[] | null>([]);
+
+  React.useEffect(() => {
+    const getInstalledRepos = async () => {
+      setLoading(true);
+      setinstalledRepos([]);
+      const res = await getInstalledRepositories();
+      setinstalledRepos(res.list);
```
Comment (Extheoisah) [inline client/fresheyes-web/src/app/components/menu/Menu.tsx:21]:
```suggestion
      setInstalledRepos(res.list);
```

File: client/fresheyes-web/src/app/components/menu/Menu.tsx [hunk 12]
```diff
@@ -0,0 +1,66 @@
...
+      console.log(res);
+    };
+
+    getInstalledRepos();
+  }, []);
+  return (
+    <div className='relative'>
+      <button onClick={() => setOpen(!open)}>
+        <picture className='flex  h-full items-end justify-end'>
+          <Image src={image || "/vercel.svg"} alt='Profile Picture' className='rounded-full' width={48} height={48} priority />
```
Comment (Extheoisah) [inline client/fresheyes-web/src/app/components/menu/Menu.tsx:32]: I can't find "vercel.svg" file anywhere in the code

File: client/fresheyes-web/src/app/components/menu/Menu.tsx [hunk 13]
```diff
@@ -0,0 +1,66 @@
...
+        </picture>
+      </button>
+
+      {open && (
+        <section className='bg-[#2d2d2d] p-4 py-5 rounded-md whitespace-nowrap flex flex-col gap-4 absolute right-0 top-14 min-w-[270px]'>
+          <Link href={"/home"} className=' hover:underline hover:font-bold flex items-center gap-2'>
```
Comment (Extheoisah) [inline client/fresheyes-web/src/app/components/menu/Menu.tsx:38]:
```suggestion
          <Link href={"/home"} className='hover:underline hover:font-bold flex items-center gap-2'>
```

File: client/fresheyes-web/src/app/components/menu/Menu.tsx [hunk 14]
```diff
@@ -0,0 +1,66 @@
...
+            Home <ExitArrowIcon />
+          </Link>
+          <button onClick={() => signOut({ callbackUrl: "/" })} className='flex hover:underline hover:font-bold items-center gap-2'>
+            Sign Out <ExitArrowIcon />
+          </button>
+          <div>
+            <p className=' underline font-semibold  pb-1'>Connected Repos</p>
```
Comment (Extheoisah) [inline client/fresheyes-web/src/app/components/menu/Menu.tsx:45]:
```suggestion
            <p className='underline font-semibold  pb-1'>Connected Repos</p>
```

File: client/fresheyes-web/src/app/components/menu/Menu.tsx [hunk 15]
```diff
@@ -0,0 +1,66 @@
...
+            ) : (
+              <>
+                {installedRepos && installedRepos?.length > 0 ? (
+                  <ol className='list-disc list-inside flex flex-col gap-1'>
+                    {installedRepos?.map((item) => (
+                      <li key={item.id}>{item.full_name}</li>
+                    ))}
+                  </ol>
+                ) : (
+                  <p>FreshEyes bot has not been connected on any of your repositories</p>
```
Comment (Extheoisah) [inline client/fresheyes-web/src/app/components/menu/Menu.tsx:57]:
```suggestion
                  <p>FreshEyes bot has not been installed in any of your repositories</p>
```

**Review by Extheoisah — CHANGES_REQUESTED** [review]
Nice work!
Just a few nits and comments.

**Comment by aassoiants** [comment]
> There is no indication that when you click on the avatar, it opens the popover. I was thinking of adding a small arrow icon to make it obvious that it's clickable?
Circle avatars are a common mental model so i think it's fine
image

> What do you think of the copy above the suggestion cards?
"Clone some of the repos below to see FreshEyes in action" 

The CTA for each of the suggested repos should read "Clone," not "Run"

Great work!

**Comment by IgboPharaoh** [comment]
will make necessary corrections for changes requested

**Comment by rejeses** [comment]
> 1. There is no indication that when you click on the avatar, it opens the popover. I was thinking of adding a small arrow icon to make it obvious that it's clickable?

This is a good idea; it could provide a visual impression that enhances the user experience

> 2. What do you think of the copy above the suggestion cards?

Also, great innovation. I'm in support.

File: client/fresheyes-web/src/app/components/form/FormSection.tsx (hunk 5 above)
Comment (IgboPharaoh) [reply client/fresheyes-web/src/app/components/form/FormSection.tsx:193]: I'm making everything here to be in uppercase but corrections are duly noted

File: client/fresheyes-web/src/app/components/form/bot.ts (hunk 6 above)
Comment (IgboPharaoh) [reply client/fresheyes-web/src/app/components/form/bot.ts:32]: I'll add a fall back in the config files

**Comment by Extheoisah** [comment]
> > 1. There is no indication that when you click on the avatar, it opens the popover. I was thinking of adding a small arrow icon to make it obvious that it's clickable?
> 
> This is a good idea; it could provide a visual impression that enhances the user experience
> 
> > 2. What do you think of the copy above the suggestion cards?
> 
> Also, great innovation. I'm in support.
> 
> A few other things not directly related to the above:
> 
> * “Pull Requests” at top left corner looks more like a clickable button than a header.
> * “Clone the PR to your GitHub account and install the FreshEyes bot to focus on the code, not the comments.”, could be adjusted to: “Clone the PR to your GitHub account and install the FreshEyes bot to focus on the code, _and not on the comments._”
> * I thought that the image at the right side was supposed to be of something else, instead of an image of the login page.
> * Possible for the “Github” in the CTA button just below the intro comments be changed to “GitHub” for uniformity?

> “Pull Requests” at top left corner looks more like a clickable button than a header.

I agree. What do you suggest is a better design choice? IMO, we should remove it.

**Comment by rejeses** [comment]
> I agree. What do you suggest is a better design choice? IMO, we should remove it.

Fully in support to be removed, as it really isn't relevant IMO

**Comment by IgboPharaoh** [comment]
> > 1. There is no indication that when you click on the avatar, it opens the popover. I was thinking of adding a small arrow icon to make it obvious that it's clickable?
> 
> This is a good idea; it could provide a visual impression that enhances the user experience
> 
> > 2. What do you think of the copy above the suggestion cards?
> 
> Also, great innovation. I'm in support.
> 
> A few other things not directly related to the above:
> 
> * “Pull Requests” at top left corner looks more like a clickable button than a header.
> * “Clone the PR to your GitHub account and install the FreshEyes bot to focus on the code, not the comments.”, could be adjusted to: “Clone the PR to your GitHub account and install the FreshEyes bot to focus on the code, _and not on the comments._”
> * I thought that the image at the right side was supposed to be of something else, instead of an image of the login page.
> * Possible for the “Github” in the CTA button just below the intro comments be changed to “GitHub” for uniformity?

- Comment 1: I think why it may not look clickable is because it's in uppercase for emphasis, but we can capitalize it to reduce the emphasis and make it look more of a title. It is very relevant and I do not support it to be removed. 
pic attached below.

image

- Comment 2: Corrections taken for 2, I agree.
- Comment 3: I think the image on the RHS isn't that of the login page but the application, though I in support we can get something better.
- Comment 4: Replacing the text formatting for `Github` alone removes the uniformity of the buttons, also we're referring to a link and not a title text here

**Comment by Extheoisah** [comment]
We were referring to the pull request button in the home page. All other things should be left as is.

Screenshot_20240411_093601_Samsung Internet

**Comment by IgboPharaoh** [comment]
Ohkkk my mistake, a misconception on my part. I'll remove the necessary items

**Review by Extheoisah — APPROVED** [review]

**Pull request merged**